File: Python/even_divisors.py
import math

if __name__ == '__main__':
    number = int(input('Enter a number: '))
    if number % 2 != 0:
        print('No even divisors!')
    else:
        count = 0
        result = []
        for i in range(1, int(math.sqrt(number)) + 1):
            if number % i == 0:
                if i % 2 == 0:
                    result.append(i)
                    count += 1
                other = number // i
                if number % other == 0 and other != i:
                    result.append(other)
                    count += 1
        print('count:', count)
        print('Even divisors: ', sorted(result))

File: Java/EvenDivisors.java
import java.util.Scanner;

public class EvenDivisors {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter any integer:");
        int n = sc.nextInt();
        sc.close();
        if (n % 2 != 0)
            System.out.println(0);
        else {
            int count = 0;
            for (int i = 1; i <= Math.sqrt(n); i++) {
                if (n % i == 0) {
                    if (i % 2 == 0)
                        count++;
                    int j = n / i;
                    if (j != i && j % 2 == 0)
                        count++;
                }
            }
            System.out.println(count);
        }
    }
}